packages/beachball/src/changelog/writeChangelog.ts

```diff
@@ -1,35 +1,121 @@
 import path from 'path';
 import fs from 'fs';
+import _ from 'lodash';
 import { ChangeSet } from '../types/ChangeInfo';
-import { PackageInfo } from '../types/PackageInfo';
-import { getPackageChangelogs } from "./getPackageChangelogs";
-import { renderChangelog } from "./renderChangelog";
-import { renderJsonChangelog } from "./renderJsonChangelog";
-export function writeChangelog(changeSet: ChangeSet, packageInfos: {
-  [pkg: string]: PackageInfo;
-}) {
+import { PackageInfo, PackageInfos } from '../types/PackageInfo';
+import { getPackageChangelogs } from './getPackageChangelogs';
+import { renderChangelog } from './renderChangelog';
+import { renderJsonChangelog } from './renderJsonChangelog';
+import { BeachballOptions } from '../types/BeachballOptions';
+import { isPathIncluded } from '../monorepo/utils';
+import { PackageChangelog } from '../types/ChangeLog';
+import { mergeChangelogs } from './mergeChangelogs';
+
+export function writeChangelog(
+  options: BeachballOptions,
+  changeSet: ChangeSet,
+  packageInfos: {
+    [pkg: string]: PackageInfo;
+  }
+): void {
+  const groupedChangelogPaths = writeGroupedChangelog(options, changeSet, packageInfos);
+  const groupedChangelogPathSet = new Set(groupedChangelogPaths);
+
   const changelogs = getPackageChangelogs(changeSet, packageInfos);
   Object.keys(changelogs).forEach(pkg => {
     const packagePath = path.dirname(packageInfos[pkg].packageJsonPath);
-    if (changelogs[pkg].comments.major ||
-      changelogs[pkg].comments.minor ||
-      changelogs[pkg].comments.patch ||
-      changelogs[pkg].comments.prerelease) {
-      const changelogFile = path.join(packagePath, 'CHANGELOG.md');
-      const previousContent = fs.existsSync(changelogFile) ? fs.readFileSync(changelogFile).toString() : '';
-      const nextContent = renderChangelog(previousContent, changelogs[pkg]);
-      fs.writeFileSync(changelogFile, nextContent);
+    if (groupedChangelogPathSet?.has(packagePath)) {
+      console.log(`Skip writing change log to ${packagePath} because grouped change log has generated here.`);
```

Comment:
If I read this right, I think the comment should be this, maybe?

```suggestion
      console.log(`Changelog for ${pkg} has been written as a group here: ${packagePath}`);
```